[libcxxabi] build failure

[cmakeerr.log](https://github.com/user-attachments/files/18071761/cmakeerr.log)
[cmakeout.log](https://github.com/user-attachments/files/18071760/cmakeout.log)
[cmakeruntimeerr.log](https://github.com/user-attachments/files/18071758/cmakeruntimeerr.log)
[cmakeruntimeout.log](https://github.com/user-attachments/files/18071759/cmakeruntimeout.log)
[ninjaout.log](https://github.com/user-attachments/files/18071763/ninjaout.log)
command：
```
cmake -S llvm -B build -DCMAKE_TOOLCHAIN_FILE=F:/Scoop/apps/vcpkg/current/scripts/buildsystems/vcpkg.cmake -DLLVM_ENABLE_PROJECTS=all -DCMAKE_BUILD_TYPE=MinSizeRel -G Ninja -DCMAKE_C_COMPILER=H:/llvm/bin/clang-cl.exe -DCMAKE_C_COMPILER_TARGET=x86_64-pc-windows-msvc -DCMAKE_CXX_COMPILER=H:/llvm/bin/clang-cl.exe -DCMAKE_CXX_COMPILER_TARGET=x86_64-pc-windows-msvc -DCMAKE_INSTALL_PREFIX="H:/llvm-project-llvmorg-19.1.5/install" -DLLVM_ENABLE_RUNTIMES="libcxxabi;libcxx"  -DLIBCXXABI_USE_LLVM_UNWINDER=off -DLIBCXX_CXX_ABI=libcxxabi -DLIBCXXABI_ENABLE_SHARED=off 
```
```
FAILED: libcxxabi/src/CMakeFiles/cxxabi_static_objects.dir/cxa_guard.cpp.obj 
H:\llvm-project-llvmorg-19.1.5\build\bin\clang-cl.exe --target=x86_64-pc-windows-msvc  /nologo -TP -DLIBCXX_BUILDING_LIBCXXABI -DUNICODE -D_CRT_NONSTDC_NO_DEPRECATE -D_CRT_NONSTDC_NO_WARNINGS -D_CRT_SECURE_NO_DEPRECATE -D_CRT_SECURE_NO_WARNINGS -D_LIBCPP_BUILDING_LIBRARY -D_LIBCPP_DISABLE_VISIBILITY_ANNOTATIONS="" -D_LIBCPP_HAS_NO_PRAGMA_SYSTEM_HEADER -D_LIBCXXABI_BUILDING_LIBRARY -D_LIBCXXABI_DISABLE_VISIBILITY_ANNOTATIONS -D_SCL_SECURE_NO_DEPRECATE -D_SCL_SECURE_NO_WARNINGS -D_UNICODE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -IH:\llvm-project-llvmorg-19.1.5\libcxxabi\..\libcxx\src -IH:\llvm-project-llvmorg-19.1.5\build\include\c++\v1 -IH:\llvm-project-llvmorg-19.1.5\libcxxabi\include /DWIN32 /D_WINDOWS   /Zc:inline /Zc:__cplusplus /Oi /Brepro /bigobj /permissive- -Werror=unguarded-availability-new /W4 -wd4141 -wd4146 -wd4244 -wd4267 -wd4291 -wd4351 -wd4456 -wd4457 -wd4458 -wd4459 -wd4503 -wd4624 -wd4722 -wd4100 -wd4127 -wd4512 -wd4505 -wd4610 -wd4510 -wd4702 -wd4245 -wd4706 -wd4310 -wd4701 -wd4703 -wd4389 -wd4611 -wd4805 -wd4204 -wd4577 -wd4091 -wd4592 -wd4319 -wd4709 -wd5105 -wd4324 -w14062 -we4238 /Gw  /O1 /Ob1 /DNDEBUG -clang:-std=c++23 -MD -fstrict-aliasing -EHsc -W4 -Wextra -Wnewline-eof -Wshadow -Wwrite-strings -Wno-unused-parameter -Wno-long-long -Werror=return-type -Wextra-semi -Wundef -Wunused-template -Wformat-nonliteral -Wno-user-defined-literals -Wno-covered-switch-default -Wno-suggest-override -Wno-error /showIncludes /Folibcxxabi\src\CMakeFiles\cxxabi_static_objects.dir\cxa_guard.cpp.obj /Fdlibcxxabi\src\CMakeFiles\cxxabi_static_objects.dir\ -c -- H:\llvm-project-llvmorg-19.1.5\libcxxabi\src\cxa_guard.cpp
In file included from H:\llvm-project-llvmorg-19.1.5\libcxxabi\src\cxa_guard.cpp:15:
In file included from H:\llvm-project-llvmorg-19.1.5\libcxxabi\src\cxa_guard_impl.h:48:
In file included from H:\llvm-project-llvmorg-19.1.5\libcxxabi\..\libcxx\src\include/atomic_support.h:13:
In file included from H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\memory:944:
In file included from H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__memory/inout_ptr.h:16:
In file included from H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__memory/shared_ptr.h:54:
In file included from H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\typeinfo:71:
D:\VisualStudio\VC\Tools\MSVC\14.42.34433\include\vcruntime_typeinfo.h(137,30): error: target of using declaration conflicts with declaration already in scope
  137 |     _VCRT_EXPORT_STD using ::type_info;
      |                              ^
D:\VisualStudio\VC\Tools\MSVC\14.42.34433\include\vcruntime_typeinfo.h(74,24): note: target of using declaration
   74 | _VCRT_EXPORT_STD class type_info // Exported because for typeid, MSVC looks for type_info in the global namespace
      |                        ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\cxxabi.h(30,27): note: conflicting declaration
   30 | class _LIBCXXABI_TYPE_VIS type_info; // forward declaration
      |                           ^
In file included from H:\llvm-project-llvmorg-19.1.5\libcxxabi\src\cxa_guard.cpp:15:
In file included from H:\llvm-project-llvmorg-19.1.5\libcxxabi\src\cxa_guard_impl.h:48:
In file included from H:\llvm-project-llvmorg-19.1.5\libcxxabi\..\libcxx\src\include/atomic_support.h:13:
In file included from H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\memory:944:
In file included from H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__memory/inout_ptr.h:16:
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__memory/shared_ptr.h(220,14): error: invalid operands to binary expression ('const type_info' and 'const std::type_info')
  220 |   return __t == typeid(_Dp) ? std::addressof(__data_.first().second()) : nullptr;
      |          ~~~ ^  ~~~~~~~~~~~
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__memory/allocator.h(253,1): note: candidate template ignored: could not match 'const allocator<_Tp>' against 'const type_info'
  253 | operator==(const allocator<_Tp>&, const allocator<_Up>&) _NOEXCEPT {
      | ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__memory/allocator.h(253,1): note: candidate template ignored: could not match 'const allocator<_Tp>' against 'const std::type_info'
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__utility/pair.h(491,1): note: candidate template ignored: could not match 'const pair<_T1, _T2>' against 'const type_info'
  491 | operator==(const pair<_T1, _T2>& __x, const pair<_U1, _U2>& __y) {
      | ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__utility/pair.h(491,1): note: candidate template ignored: could not match 'const pair<_T1, _T2>' against 'const std::type_info'
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__iterator/reverse_iterator.h(206,1): note: candidate template ignored: could not match 'const reverse_iterator<_Iter1>' against 'const type_info'
  206 | operator==(const reverse_iterator<_Iter1>& __x, const reverse_iterator<_Iter2>& __y)
      | ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__memory/unique_ptr.h(496,1): note: candidate template ignored: could not match 'const unique_ptr<_T1, _D1>' against 'const type_info'
  496 | operator==(const unique_ptr<_T1, _D1>& __x, const unique_ptr<_T2, _D2>& __y) {
      | ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__memory/unique_ptr.h(496,1): note: candidate template ignored: could not match 'const unique_ptr<_T1, _D1>' against 'const std::type_info'
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__memory/unique_ptr.h(542,1): note: candidate template ignored: could not match 'const unique_ptr<_T1, _D1>' against 'const type_info'
  542 | operator==(const unique_ptr<_T1, _D1>& __x, nullptr_t) _NOEXCEPT {
      | ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__memory/unique_ptr.h(542,1): note: candidate template ignored: could not match 'const unique_ptr<_T1, _D1>' against 'const std::type_info'
2 errors generated.
[1099/1291] H:\llvm-project-llvmorg-19.1.5\build\bin\clang-cl.exe --target=x86_64-pc-windows-msvc  /nologo -TP -DLIBCXX_BUILDING_LIBCXXABI -DUNICODE -D_CRT_NONSTDC_NO_DEPRECATE -D_CRT_NONSTDC_NO_WARNINGS -D_CRT_SECURE_NO_DEPRECATE -D_CRT_SECURE_NO_WARNINGS -D_LIBCPP_BUILDING_LIBRARY -D_LIBCPP_DISABLE_VISIBILITY_ANNOTATIONS="" -D_LIBCPP_HAS_NO_PRAGMA_SYSTEM_HEADER -D_LIBCXXABI_BUILDING_LIBRARY -D_LIBCXXABI_DISABLE_VISIBILITY_ANNOTATIONS -D_SCL_SECURE_NO_DEPRECATE -D_SCL_SECURE_NO_WARNINGS -D_UNICODE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -IH:\llvm-project-llvmorg-19.1.5\libcxxabi\..\libcxx\src -IH:\llvm-project-llvmorg-19.1.5\build\include\c++\v1 -IH:\llvm-project-llvmorg-19.1.5\libcxxabi\include /DWIN32 /D_WINDOWS   /Zc:inline /Zc:__cplusplus /Oi /Brepro /bigobj /permissive- -Werror=unguarded-availability-new /W4 -wd4141 -wd4146 -wd4244 -wd4267 -wd4291 -wd4351 -wd4456 -wd4457 -wd4458 -wd4459 -wd4503 -wd4624 -wd4722 -wd4100 -wd4127 -wd4512 -wd4505 -wd4610 -wd4510 -wd4702 -wd4245 -wd4706 -wd4310 -wd4701 -wd4703 -wd4389 -wd4611 -wd4805 -wd4204 -wd4577 -wd4091 -wd4592 -wd4319 -wd4709 -wd5105 -wd4324 -w14062 -we4238 /Gw  /O1 /Ob1 /DNDEBUG -clang:-std=c++23 -MD -fstrict-aliasing -EHsc -W4 -Wextra -Wnewline-eof -Wshadow -Wwrite-strings -Wno-unused-parameter -Wno-long-long -Werror=return-type -Wextra-semi -Wundef -Wunused-template -Wformat-nonliteral -Wno-user-defined-literals -Wno-covered-switch-default -Wno-suggest-override -Wno-error /showIncludes /Folibcxxabi\src\CMakeFiles\cxxabi_static_objects.dir\cxa_demangle.cpp.obj /Fdlibcxxabi\src\CMakeFiles\cxxabi_static_objects.dir\ -c -- H:\llvm-project-llvmorg-19.1.5\libcxxabi\src\cxa_demangle.cpp
FAILED: libcxxabi/src/CMakeFiles/cxxabi_static_objects.dir/cxa_demangle.cpp.obj 
H:\llvm-project-llvmorg-19.1.5\build\bin\clang-cl.exe --target=x86_64-pc-windows-msvc  /nologo -TP -DLIBCXX_BUILDING_LIBCXXABI -DUNICODE -D_CRT_NONSTDC_NO_DEPRECATE -D_CRT_NONSTDC_NO_WARNINGS -D_CRT_SECURE_NO_DEPRECATE -D_CRT_SECURE_NO_WARNINGS -D_LIBCPP_BUILDING_LIBRARY -D_LIBCPP_DISABLE_VISIBILITY_ANNOTATIONS="" -D_LIBCPP_HAS_NO_PRAGMA_SYSTEM_HEADER -D_LIBCXXABI_BUILDING_LIBRARY -D_LIBCXXABI_DISABLE_VISIBILITY_ANNOTATIONS -D_SCL_SECURE_NO_DEPRECATE -D_SCL_SECURE_NO_WARNINGS -D_UNICODE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -IH:\llvm-project-llvmorg-19.1.5\libcxxabi\..\libcxx\src -IH:\llvm-project-llvmorg-19.1.5\build\include\c++\v1 -IH:\llvm-project-llvmorg-19.1.5\libcxxabi\include /DWIN32 /D_WINDOWS   /Zc:inline /Zc:__cplusplus /Oi /Brepro /bigobj /permissive- -Werror=unguarded-availability-new /W4 -wd4141 -wd4146 -wd4244 -wd4267 -wd4291 -wd4351 -wd4456 -wd4457 -wd4458 -wd4459 -wd4503 -wd4624 -wd4722 -wd4100 -wd4127 -wd4512 -wd4505 -wd4610 -wd4510 -wd4702 -wd4245 -wd4706 -wd4310 -wd4701 -wd4703 -wd4389 -wd4611 -wd4805 -wd4204 -wd4577 -wd4091 -wd4592 -wd4319 -wd4709 -wd5105 -wd4324 -w14062 -we4238 /Gw  /O1 /Ob1 /DNDEBUG -clang:-std=c++23 -MD -fstrict-aliasing -EHsc -W4 -Wextra -Wnewline-eof -Wshadow -Wwrite-strings -Wno-unused-parameter -Wno-long-long -Werror=return-type -Wextra-semi -Wundef -Wunused-template -Wformat-nonliteral -Wno-user-defined-literals -Wno-covered-switch-default -Wno-suggest-override -Wno-error /showIncludes /Folibcxxabi\src\CMakeFiles\cxxabi_static_objects.dir\cxa_demangle.cpp.obj /Fdlibcxxabi\src\CMakeFiles\cxxabi_static_objects.dir\ -c -- H:\llvm-project-llvmorg-19.1.5\libcxxabi\src\cxa_demangle.cpp
In file included from H:\llvm-project-llvmorg-19.1.5\libcxxabi\src\cxa_demangle.cpp:23:
In file included from H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\exception:81:
In file included from H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__exception/exception_ptr.h:20:
In file included from H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\typeinfo:71:
D:\VisualStudio\VC\Tools\MSVC\14.42.34433\include\vcruntime_typeinfo.h(137,30): error: target of using declaration conflicts with declaration already in scope
  137 |     _VCRT_EXPORT_STD using ::type_info;
      |                              ^
D:\VisualStudio\VC\Tools\MSVC\14.42.34433\include\vcruntime_typeinfo.h(74,24): note: target of using declaration
   74 | _VCRT_EXPORT_STD class type_info // Exported because for typeid, MSVC looks for type_info in the global namespace
      |                        ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\cxxabi.h(30,27): note: conflicting declaration
   30 | class _LIBCXXABI_TYPE_VIS type_info; // forward declaration
      |                           ^
In file included from H:\llvm-project-llvmorg-19.1.5\libcxxabi\src\cxa_demangle.cpp:24:
In file included from H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\functional:548:
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__functional/function.h(320,12): error: invalid operands to binary expression ('const type_info' and 'const std::type_info')
  320 |   if (__ti == typeid(_Fp))
      |       ~~~~ ^  ~~~~~~~~~~~
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__exception/exception_ptr.h(152,32): note: candidate function not viable: cannot convert argument of incomplete type 'const type_info' to 'const exception_ptr' for 1st argument
  152 | _LIBCPP_EXPORTED_FROM_ABI bool operator==(const exception_ptr& __x, const exception_ptr& __y) _NOEXCEPT;
      |                                ^          ~~~~~~~~~~~~~~~~~~~~~~~~
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__utility/pair.h(491,1): note: candidate template ignored: could not match 'const pair<_T1, _T2>' against 'const type_info'
  491 | operator==(const pair<_T1, _T2>& __x, const pair<_U1, _U2>& __y) {
      | ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__utility/pair.h(491,1): note: candidate template ignored: could not match 'const pair<_T1, _T2>' against 'const std::type_info'
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__iterator/reverse_iterator.h(206,1): note: candidate template ignored: could not match 'const reverse_iterator<_Iter1>' against 'const type_info'
  206 | operator==(const reverse_iterator<_Iter1>& __x, const reverse_iterator<_Iter2>& __y)
      | ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__iterator/wrap_iter.h(114,1): note: candidate template ignored: could not match 'const __wrap_iter<_Iter1>' against 'const type_info'
  114 | operator==(const __wrap_iter<_Iter1>& __x, const __wrap_iter<_Iter1>& __y) _NOEXCEPT {
      | ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__iterator/wrap_iter.h(120,1): note: candidate template ignored: could not match 'const __wrap_iter<_Iter1>' against 'const type_info'
  120 | operator==(const __wrap_iter<_Iter1>& __x, const __wrap_iter<_Iter2>& __y) _NOEXCEPT {
      | ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__iterator/wrap_iter.h(120,1): note: candidate template ignored: could not match 'const __wrap_iter<_Iter1>' against 'const std::type_info'
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\string_view(707,38): note: candidate template ignored: could not match 'basic_string_view<_CharT, _Traits>' against 'type_info'
  707 | _LIBCPP_HIDE_FROM_ABI constexpr bool operator==(basic_string_view<_CharT, _Traits> __lhs,
      |                                      ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\string_view(707,38): note: candidate template ignored: could not match 'basic_string_view<_CharT, _Traits>' against 'std::type_info'
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\array(389,1): note: candidate template ignored: could not match 'const array<_Tp, _Size>' against 'const type_info'
  389 | operator==(const array<_Tp, _Size>& __x, const array<_Tp, _Size>& __y) {
      | ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\optional(987,1): note: candidate template ignored: could not match 'const optional<_Tp>' against 'const type_info'
  987 | operator==(const optional<_Tp>& __x, const optional<_Up>& __y) {
      | ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\optional(1069,38): note: candidate template ignored: could not match 'const optional<_Tp>' against 'const type_info'
 1069 | _LIBCPP_HIDE_FROM_ABI constexpr bool operator==(const optional<_Tp>& __x, nullopt_t) noexcept {
      |                                      ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\optional(1069,38): note: candidate template ignored: could not match 'const optional<_Tp>' against 'const std::type_info'
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\optional(1144,1): note: candidate template ignored: could not match 'const optional<_Tp>' against 'const type_info'
 1144 | operator==(const optional<_Tp>& __x, const _Up& __v) {
      | ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\optional(1152,1): note: candidate template ignored: could not match 'const optional<_Up>' against 'const std::type_info'
 1152 | operator==(const _Tp& __v, const optional<_Up>& __x) {
      | ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__memory/unique_ptr.h(496,1): note: candidate template ignored: could not match 'const unique_ptr<_T1, _D1>' against 'const type_info'
  496 | operator==(const unique_ptr<_T1, _D1>& __x, const unique_ptr<_T2, _D2>& __y) {
      | ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__memory/unique_ptr.h(496,1): note: candidate template ignored: could not match 'const unique_ptr<_T1, _D1>' against 'const std::type_info'
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__memory/unique_ptr.h(542,1): note: candidate template ignored: could not match 'const unique_ptr<_T1, _D1>' against 'const type_info'
  542 | operator==(const unique_ptr<_T1, _D1>& __x, nullptr_t) _NOEXCEPT {
      | ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__memory/unique_ptr.h(542,1): note: candidate template ignored: could not match 'const unique_ptr<_T1, _D1>' against 'const std::type_info'
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\tuple(1150,1): note: candidate template ignored: could not match 'const tuple<_Tp...>' against 'const type_info'
 1150 | operator==(const tuple<_Tp...>& __x, const tuple<_Up...>& __y) {
      | ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\tuple(1150,1): note: candidate template ignored: could not match 'const tuple<_Tp...>' against 'const std::type_info'
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__memory/allocator.h(253,1): note: candidate template ignored: could not match 'const allocator<_Tp>' against 'const type_info'
  253 | operator==(const allocator<_Tp>&, const allocator<_Up>&) _NOEXCEPT {
      | ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__memory/allocator.h(253,1): note: candidate template ignored: could not match 'const allocator<_Tp>' against 'const std::type_info'
In file included from H:\llvm-project-llvmorg-19.1.5\libcxxabi\src\cxa_demangle.cpp:24:
In file included from H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\functional:548:
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__functional/function.h(732,45): error: invalid operands to binary expression ('const std::type_info' and 'const std::type_info')
  732 |     if (__policy_->__is_null || typeid(_Tp) != *__policy_->__type_info)
      |                                 ~~~~~~~~~~~ ^  ~~~~~~~~~~~~~~~~~~~~~~~
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__exception/exception_ptr.h(152,32): note: candidate function not viable: cannot convert argument of incomplete type 'const std::type_info' to 'const exception_ptr' for 1st argument
  152 | _LIBCPP_EXPORTED_FROM_ABI bool operator==(const exception_ptr& __x, const exception_ptr& __y) _NOEXCEPT;
      |                                ^          ~~~~~~~~~~~~~~~~~~~~~~~~
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__exception/exception_ptr.h(154,35): note: candidate function not viable: cannot convert argument of incomplete type 'const std::type_info' to 'const exception_ptr' for 1st argument
  154 | inline _LIBCPP_HIDE_FROM_ABI bool operator!=(const exception_ptr& __x, const exception_ptr& __y) _NOEXCEPT {
      |                                   ^          ~~~~~~~~~~~~~~~~~~~~~~~~
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__utility/pair.h(491,1): note: candidate template ignored: could not match 'const pair<_T1, _T2>' against 'const std::type_info'
  491 | operator==(const pair<_T1, _T2>& __x, const pair<_U1, _U2>& __y) {
      | ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__utility/pair.h(491,1): note: candidate template ignored: could not match 'const pair<_T1, _T2>' against 'const std::type_info'
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__iterator/reverse_iterator.h(206,1): note: candidate template ignored: could not match 'const reverse_iterator<_Iter1>' against 'const std::type_info'
  206 | operator==(const reverse_iterator<_Iter1>& __x, const reverse_iterator<_Iter2>& __y)
      | ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__iterator/reverse_iterator.h(230,1): note: candidate template ignored: could not match 'const reverse_iterator<_Iter1>' against 'const std::type_info'
  230 | operator!=(const reverse_iterator<_Iter1>& __x, const reverse_iterator<_Iter2>& __y)
      | ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__iterator/wrap_iter.h(114,1): note: candidate template ignored: could not match 'const __wrap_iter<_Iter1>' against 'const std::type_info'
  114 | operator==(const __wrap_iter<_Iter1>& __x, const __wrap_iter<_Iter1>& __y) _NOEXCEPT {
      | ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__iterator/wrap_iter.h(120,1): note: candidate template ignored: could not match 'const __wrap_iter<_Iter1>' against 'const std::type_info'
  120 | operator==(const __wrap_iter<_Iter1>& __x, const __wrap_iter<_Iter2>& __y) _NOEXCEPT {
      | ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__iterator/wrap_iter.h(120,1): note: candidate template ignored: could not match 'const __wrap_iter<_Iter1>' against 'const std::type_info'
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\string_view(707,38): note: candidate template ignored: could not match 'basic_string_view<_CharT, _Traits>' against 'std::type_info'
  707 | _LIBCPP_HIDE_FROM_ABI constexpr bool operator==(basic_string_view<_CharT, _Traits> __lhs,
      |                                      ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\string_view(707,38): note: candidate template ignored: could not match 'basic_string_view<_CharT, _Traits>' against 'std::type_info'
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\array(389,1): note: candidate template ignored: could not match 'const array<_Tp, _Size>' against 'const std::type_info'
  389 | operator==(const array<_Tp, _Size>& __x, const array<_Tp, _Size>& __y) {
      | ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\optional(987,1): note: candidate template ignored: could not match 'const optional<_Tp>' against 'const std::type_info'
  987 | operator==(const optional<_Tp>& __x, const optional<_Up>& __y) {
      | ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\optional(999,1): note: candidate template ignored: could not match 'const optional<_Tp>' against 'const std::type_info'
  999 | operator!=(const optional<_Tp>& __x, const optional<_Up>& __y) {
      | ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\optional(1069,38): note: candidate template ignored: could not match 'const optional<_Tp>' against 'const std::type_info'
 1069 | _LIBCPP_HIDE_FROM_ABI constexpr bool operator==(const optional<_Tp>& __x, nullopt_t) noexcept {
      |                                      ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\optional(1069,38): note: candidate template ignored: could not match 'const optional<_Tp>' against 'const std::type_info'
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\optional(1144,1): note: candidate template ignored: could not match 'const optional<_Tp>' against 'const std::type_info'
 1144 | operator==(const optional<_Tp>& __x, const _Up& __v) {
      | ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\optional(1152,1): note: candidate template ignored: could not match 'const optional<_Up>' against 'const std::type_info'
 1152 | operator==(const _Tp& __v, const optional<_Up>& __x) {
      | ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\optional(1160,1): note: candidate template ignored: could not match 'const optional<_Tp>' against 'const std::type_info'
 1160 | operator!=(const optional<_Tp>& __x, const _Up& __v) {
      | ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\optional(1168,1): note: candidate template ignored: could not match 'const optional<_Up>' against 'const std::type_info'
 1168 | operator!=(const _Tp& __v, const optional<_Up>& __x) {
      | ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__memory/unique_ptr.h(496,1): note: candidate template ignored: could not match 'const unique_ptr<_T1, _D1>' against 'const std::type_info'
  496 | operator==(const unique_ptr<_T1, _D1>& __x, const unique_ptr<_T2, _D2>& __y) {
      | ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__memory/unique_ptr.h(496,1): note: candidate template ignored: could not match 'const unique_ptr<_T1, _D1>' against 'const std::type_info'
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__memory/unique_ptr.h(542,1): note: candidate template ignored: could not match 'const unique_ptr<_T1, _D1>' against 'const std::type_info'
  542 | operator==(const unique_ptr<_T1, _D1>& __x, nullptr_t) _NOEXCEPT {
      | ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__memory/unique_ptr.h(542,1): note: candidate template ignored: could not match 'const unique_ptr<_T1, _D1>' against 'const std::type_info'
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\tuple(1150,1): note: candidate template ignored: could not match 'const tuple<_Tp...>' against 'const std::type_info'
 1150 | operator==(const tuple<_Tp...>& __x, const tuple<_Up...>& __y) {
      | ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\tuple(1150,1): note: candidate template ignored: could not match 'const tuple<_Tp...>' against 'const std::type_info'
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__memory/allocator.h(253,1): note: candidate template ignored: could not match 'const allocator<_Tp>' against 'const std::type_info'
  253 | operator==(const allocator<_Tp>&, const allocator<_Up>&) _NOEXCEPT {
      | ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__memory/allocator.h(253,1): note: candidate template ignored: could not match 'const allocator<_Tp>' against 'const std::type_info'
In file included from H:\llvm-project-llvmorg-19.1.5\libcxxabi\src\cxa_demangle.cpp:24:
In file included from H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\functional:552:
In file included from H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__functional/boyer_moore_searcher.h:22:
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__memory/shared_ptr.h(220,14): error: invalid operands to binary expression ('const type_info' and 'const std::type_info')
  220 |   return __t == typeid(_Dp) ? std::addressof(__data_.first().second()) : nullptr;
      |          ~~~ ^  ~~~~~~~~~~~
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__exception/exception_ptr.h(152,32): note: candidate function not viable: cannot convert argument of incomplete type 'const type_info' to 'const exception_ptr' for 1st argument
  152 | _LIBCPP_EXPORTED_FROM_ABI bool operator==(const exception_ptr& __x, const exception_ptr& __y) _NOEXCEPT;
      |                                ^          ~~~~~~~~~~~~~~~~~~~~~~~~
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__utility/pair.h(491,1): note: candidate template ignored: could not match 'const pair<_T1, _T2>' against 'const type_info'
  491 | operator==(const pair<_T1, _T2>& __x, const pair<_U1, _U2>& __y) {
      | ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__utility/pair.h(491,1): note: candidate template ignored: could not match 'const pair<_T1, _T2>' against 'const std::type_info'
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__iterator/reverse_iterator.h(206,1): note: candidate template ignored: could not match 'const reverse_iterator<_Iter1>' against 'const type_info'
  206 | operator==(const reverse_iterator<_Iter1>& __x, const reverse_iterator<_Iter2>& __y)
      | ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__iterator/wrap_iter.h(114,1): note: candidate template ignored: could not match 'const __wrap_iter<_Iter1>' against 'const type_info'
  114 | operator==(const __wrap_iter<_Iter1>& __x, const __wrap_iter<_Iter1>& __y) _NOEXCEPT {
      | ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__iterator/wrap_iter.h(120,1): note: candidate template ignored: could not match 'const __wrap_iter<_Iter1>' against 'const type_info'
  120 | operator==(const __wrap_iter<_Iter1>& __x, const __wrap_iter<_Iter2>& __y) _NOEXCEPT {
      | ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__iterator/wrap_iter.h(120,1): note: candidate template ignored: could not match 'const __wrap_iter<_Iter1>' against 'const std::type_info'
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\string_view(707,38): note: candidate template ignored: could not match 'basic_string_view<_CharT, _Traits>' against 'type_info'
  707 | _LIBCPP_HIDE_FROM_ABI constexpr bool operator==(basic_string_view<_CharT, _Traits> __lhs,
      |                                      ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\string_view(707,38): note: candidate template ignored: could not match 'basic_string_view<_CharT, _Traits>' against 'std::type_info'
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\array(389,1): note: candidate template ignored: could not match 'const array<_Tp, _Size>' against 'const type_info'
  389 | operator==(const array<_Tp, _Size>& __x, const array<_Tp, _Size>& __y) {
      | ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\optional(987,1): note: candidate template ignored: could not match 'const optional<_Tp>' against 'const type_info'
  987 | operator==(const optional<_Tp>& __x, const optional<_Up>& __y) {
      | ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\optional(1069,38): note: candidate template ignored: could not match 'const optional<_Tp>' against 'const type_info'
 1069 | _LIBCPP_HIDE_FROM_ABI constexpr bool operator==(const optional<_Tp>& __x, nullopt_t) noexcept {
      |                                      ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\optional(1069,38): note: candidate template ignored: could not match 'const optional<_Tp>' against 'const std::type_info'
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\optional(1144,1): note: candidate template ignored: could not match 'const optional<_Tp>' against 'const type_info'
 1144 | operator==(const optional<_Tp>& __x, const _Up& __v) {
      | ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\optional(1152,1): note: candidate template ignored: could not match 'const optional<_Up>' against 'const std::type_info'
 1152 | operator==(const _Tp& __v, const optional<_Up>& __x) {
      | ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__memory/unique_ptr.h(496,1): note: candidate template ignored: could not match 'const unique_ptr<_T1, _D1>' against 'const type_info'
  496 | operator==(const unique_ptr<_T1, _D1>& __x, const unique_ptr<_T2, _D2>& __y) {
      | ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__memory/unique_ptr.h(496,1): note: candidate template ignored: could not match 'const unique_ptr<_T1, _D1>' against 'const std::type_info'
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__memory/unique_ptr.h(542,1): note: candidate template ignored: could not match 'const unique_ptr<_T1, _D1>' against 'const type_info'
  542 | operator==(const unique_ptr<_T1, _D1>& __x, nullptr_t) _NOEXCEPT {
      | ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__memory/unique_ptr.h(542,1): note: candidate template ignored: could not match 'const unique_ptr<_T1, _D1>' against 'const std::type_info'
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\tuple(1150,1): note: candidate template ignored: could not match 'const tuple<_Tp...>' against 'const type_info'
 1150 | operator==(const tuple<_Tp...>& __x, const tuple<_Up...>& __y) {
      | ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\tuple(1150,1): note: candidate template ignored: could not match 'const tuple<_Tp...>' against 'const std::type_info'
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__memory/allocator.h(253,1): note: candidate template ignored: could not match 'const allocator<_Tp>' against 'const type_info'
  253 | operator==(const allocator<_Tp>&, const allocator<_Up>&) _NOEXCEPT {
      | ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__memory/allocator.h(253,1): note: candidate template ignored: could not match 'const allocator<_Tp>' against 'const std::type_info'
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__functional/function.h(1014,35): note: candidate template ignored: could not match 'const function<_Rp (_ArgTypes...)>' against 'const type_info'
 1014 | inline _LIBCPP_HIDE_FROM_ABI bool operator==(const function<_Rp(_ArgTypes...)>& __f, nullptr_t) _NOEXCEPT {
      |                                   ^
H:\llvm-project-llvmorg-19.1.5\build\include\c++\v1\__functional/function.h(1014,35): note: candidate template ignored: could not match 'const function<_Rp (_ArgTypes...)>' against 'const std::type_info'
4 errors generated.
ninja: build stopped: subcommand failed.
```
success when libcxx is built only

--- a/libcxxabi/CMakeLists.txt
+++ b/libcxxabi/CMakeLists.txt
@@ -34,6 +34,11 @@ MACRO_ENSURE_OUT_OF_SOURCE_BUILD(
  build directory and run 'cmake /path/to/${PROJECT_NAME} [options]' there."
  )
 
+if (MSVC)
+  message(FATAL_ERROR "Libc++abi can't be built for MSVC targets, and doing so is pointless anyway because such "
+                      "targets must use the MS C++ ABI, and libc++abi provides the Itanium C++ ABI.")
+endif()
+
 #===============================================================================
 # Setup CMake Options
 #===============================================================================

--- a/libunwind/CMakeLists.txt
+++ b/libunwind/CMakeLists.txt
@@ -24,6 +24,11 @@ set(LIBUNWIND_LIBCXX_PATH "${CMAKE_CURRENT_LIST_DIR}/../libcxx" CACHE PATH
 include(GNUInstallDirs)
 include(CheckSymbolExists)
 
+if (MSVC)
+  message(FATAL_ERROR "Libunwind doesn't build for MSVC targets, and that is almost certainly not what you want to do "
+                      "anyway since libunwind is tied to the Itanium C++ ABI, and MSVC targets must use the MS C++ ABI.")
+endif()
+
 #===============================================================================
 # Setup CMake Options
 #===============================================================================

